### PR TITLE
Add $arguments for after event

### PR DESCRIPTION
**Pull request by GrahamCampbell**
Replaces #353.

### CODE REVIEW

File: src/Event/After.php [hunk 1]
```diff
@@ -89,4 +96,60 @@ public function setResult($result)
...
+     *
+     * @param string $key     argument key
+     * @param mixed  $default default return value
+     *
+     * @return mixed
+     */
+    public function getArgument($key, $default = null)
+    {
+        if (! array_key_exists($key, $this->arguments)) {
+            return $default;
```
Comment (frankdejonge) [inline src/Event/After.php]: @GrahamCampbell ping!

Comment (GrahamCampbell) [inline src/Event/After.php]: Yeh, probably. Ping @Gounlaf.

Comment (Gounlaf) [inline src/Event/After.php]: Instead of a default value, you think it would be better to throw an exception ?

Comment (frankdejonge) [inline src/Event/After.php]: @Gounlaf correct. The event should not be abused to transport parameters that don't affect the call.

**Pull request merged**